tests/testthat/test-utils.R

```diff
@@ -20,8 +22,122 @@ test_that("log scale [PMP-TEST-063]", {
 })
 
 test_that("rot_x and rot_y [PMP-TEST-064]", {
-  expect_is(p + rot_x(), "gg")
-  expect_is(p + rot_y(), "gg")
+  x <- p + rot_x(angle = 33)
+  expect_is(x, "gg")
+  expect_equal(x$theme$axis.text.x$angle, 33)
+
+  y <- p + rot_y(angle = 11)
+  expect_equal(y$theme$axis.text.y$angle, 11)
+
+  x <- p + rot_x(vertical = TRUE)
+  expect_equal(x$theme$axis.text.x$angle, 90)
+  expect_equal(x$theme$axis.text.x$hjust, 1)
+
+  y <- p + rot_y(vertical = TRUE)
+  expect_equal(y$theme$axis.text.y$angle, 90)
+  expect_equal(y$theme$axis.text.y$vjust, 1)
+
+  x1 <- p + rot_x(vertical = TRUE, hjust = "bottom")
+  expect_equal(x1$theme$axis.text.x$angle, 90)
+  expect_equal(x1$theme$axis.text.x$hjust, 0)
+
+  x2 <- p + rot_x(vertical = TRUE, hjust = "top")
+  expect_equal(x2$theme$axis.text.x$angle, 90)
+  expect_equal(x2$theme$axis.text.x$hjust, 1)
+
+  y1 <- p + rot_y(vertical = TRUE, vjust = "left")
+  expect_equal(y1$theme$axis.text.y$angle, 90)
+  expect_equal(y1$theme$axis.text.y$vjust, 1)
+
+  y2 <- p + rot_y(vertical = TRUE, vjust = "right")
+  expect_equal(y2$theme$axis.text.y$angle, 90)
+  expect_equal(y2$theme$axis.text.y$vjust, 0)
+})
+
+test_that("rot_at - rotate list of plots", {
+  etas <- paste0("ETA", 1:3)
+  co <- c("STUDYc", "CPc", "RF")
+
+  x0 <- eta_cat(id, x = co, y = etas)
+
+  expect_identical(
+    names(x0)[1:3],
+    c("ETA1vSTUDYc", "ETA2vSTUDYc", "ETA3vSTUDYc")
+  )
+
+  x <- rot_at(x0, at = "ETA2vSTUDYc", angle = 11)
+  expect_is(x, "list")
+  expect_is(x[[1]], "gg")
+  expect_equal(x$ETA2vSTUDYc$theme$axis.text.x$angle, 11)
+  expect_null(x$ETA1vSTUDYc$theme$axis.text.x$angle)
+
+  x <- rot_at(x0, at = c("ETA2vSTUDYc", "ETA1vRF"), angle = 11)
+  expect_is(x, "list")
+  expect_equal(x$ETA2vSTUDYc$theme$axis.text.x$angle, 11)
+  expect_equal(x$ETA1vRF$theme$axis.text.x$angle, 11)
+
+  x <- rot_at(x0, re = "ETA2", angle = 33)
+  expect_is(x, "list")
+  expect_is(x[[1]], "gg")
+  expect_equal(x$ETA2vSTUDYc$theme$axis.text.x$angle, 33)
+  expect_equal(x$ETA2vCPc$theme$axis.text.x$angle, 33)
+  expect_equal(x$ETA2vRF$theme$axis.text.x$angle, 33)
+  expect_null(x$ETA1vSTUDYc$theme$axis.text.x$angle)
+  expect_null(x$ETA3vCPc$theme$axis.text.x$angle)
+
+  expect_error(rot_at(x0, at = "kyle"), "requested names not found")
+  expect_warning(rot_at(x0, re = "banana"), "did not find any plots")
+  expect_error(rot_at(x0[[1]]), "must be a list of gg")
+  expect_error(rot_at(unname(x0)), "must be named")
+})
+
+test_that("rot_xy", {
+  g1 <- dv_pred(data)
+  g2 <- dv_ipred(data)
+  p1 <- g1 + g2
+  p2 <- g1/g2
+  l <- list(a = g1, b = g2)
+  lp <- list(a = p1, b = p2)
+  l0 <- unname(l)
+
+  as_img <- function(x) {
```

Comment:
These tests were failing on R < 4.1 but working fine locally and on R >= 4.1. I don't think there was an inconsistency with the plots, but couldn't test that they were equal. I think this is worked out now by just writing out a png file and comparing the hash.

Reply:
> These tests were failing on R < 4.1 but working fine locally and on R >= 4.1.

Hmm, yeah, I'm not sure where that difference is coming from.

> I don't think there was an inconsistency with the plots, but couldn't test that they were equal. I think this is worked out now by just writing out a png file and comparing the hash.

That works for me.

Fwiw the reason your `nuke_env` from the earlier commits didn't work is some of the results had `plot_env` environments that were deeper.  I confirmed that all the CI jobs pass if, before comparison, we remove all the `plot_env` components throughout the results: 7d4bd05fa8ed034530ce2002e937ecce38e9f66d

If you prefer that approach, you can bring that commit over to this PR with

```
$ git fetch origin
$ git cherry-pick 7d4bd05fa8ed034530ce2002e937ecce38e9f66d
```

Reply:
I think `nuke_env()` did ok if it was just a `gg` plot, but didn't work for patchwork and I didn't want to mess with going into that patch object; probably not too bad, but preferred a single approach to handle everything and I was losing patience with it.

Reply:
OIC ... that doesn't look as bad as I was thinking; let me try to cherry pick and I'll go with that. Thanks for digging into that ... it was driving me crazy trying to come up with a reliable test.

Reply:
This is awesome, the solution and the `cherry-pick`. Just pushed this in. Thanks!